ethereum/trie/src/main/java/tech/pegasys/pantheon/ethereum/trie/StoredMerklePatriciaTrie.java

```diff
@@ -40,12 +42,19 @@
    * @param nodeLoader The {@link NodeLoader} to retrieve node data from.
    * @param valueSerializer A function for serializing values to bytes.
    * @param valueDeserializer A function for deserializing values from bytes.
+   * @param singleUseNodes whether nodes should unload data after being visited
    */
   public StoredMerklePatriciaTrie(
       final NodeLoader nodeLoader,
       final Function<V, BytesValue> valueSerializer,
-      final Function<BytesValue, V> valueDeserializer) {
-    this(nodeLoader, MerklePatriciaTrie.EMPTY_TRIE_NODE_HASH, valueSerializer, valueDeserializer);
+      final Function<BytesValue, V> valueDeserializer,
+      final boolean singleUseNodes) {
```

Comment:
As a follow up, it would be worth seeing if we can replace this flag with something in `AllNodesVisitor` - e.g. after visiting each node it calls `Node.release` or something.  If you're iterating all nodes, you never want them to all stay in memory so it would be nice if that automatically took care of itself and then the change would be a lot more focussed as well.